### PR TITLE
remeshing triggered by minimum mesh quality #1961

### PR DESCRIPTION
**Pull request by lberti**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
- [ ] Have you successfully run the Feel++ testsuite with your changes locally?
- [ ] Have you written Doxygen comments in your contribution ?

-----

### CODE REVIEW

File: toolboxes/feel/feelmodels/modelcore/modelmeshes.hpp [hunk 1]
```diff
@@ -360,15 +373,18 @@ public :
             static tabulate_informations_ptr_t tabulateInformations( nl::json const& jsonInfo, TabulateInformationProperties const& tabInfoProp );
 
             friend struct Execute;
+            
+            double M_min_quality_mas=0.9;
```
Comment (prudhomm) [inline toolboxes/feel/feelmodels/modelcore/modelmeshes.hpp:377]: @lberti what does `mas` mean? shouldn't it be threshold ?

File: toolboxes/feel/feelmodels/modelmesh/meshale.hpp [hunk 2]
```diff
@@ -255,6 +259,8 @@ public :
 
     void applyRemesh( mesh_ptrtype const& newMesh, std::vector<std::tuple<std::string,range_elements_type>> const& rangeElt );
 
+    void attachModelMeshesptr(std::shared_ptr<ModelMesh<index_type>> const& modelMesh_ptr);
```
Comment (prudhomm) [inline toolboxes/feel/feelmodels/modelmesh/meshale.hpp:262]: rename to `attachModelMeshes` ?

File: toolboxes/fluid/cases/TurekHron/moving_body/cfd3.json [hunk 3]
```diff
@@ -74,6 +74,7 @@
                     "events":{
                         "each_time_step":{
                             "frequency":10
+                            // "min_quality":0.4
```
Comment (prudhomm) [inline toolboxes/fluid/cases/TurekHron/moving_body/cfd3.json:77]: why is it commented out ? would it be possible to add a test for this ?

**Review by prudhomm — APPROVED** [review]
@lberti it seems ok to me but there are a few update to do I think

**Review by vincentchabannes — CHANGES_REQUESTED** [review]
@lberti this is a first step but still a lot of thing to handle.
 We probably need to change the weak ptr with a raw ptr in order to avoid the shared_from_this inheritance. The observer desgin pattern should be improved, and the event mesh_quality should be handle when the mesh is moved (and even before), not in the event each_time_step. Also, I think here the mesh quality is evaluated at each time, it shouldn't , only when the json say that this event can appear. The json section should be also improved.

**Comment by prudhomm** [comment]
@vincentchabannes what do you mean by should be improved? what is wrong with the current implementation of the observer pattern ? may be I should have a look more closely and not just the diff.
regarding the computational efficiency I agree.

**Pull request closed without merging**